node/src/test/scala/com/wavesplatform/transaction/data/DataTransactionNewLimitsSpec.scala

```diff
@@ -0,0 +1,69 @@
+package com.wavesplatform.transaction.data
+
+import com.wavesplatform.common.state.ByteStr
+import com.wavesplatform.db.WithDomain
+import com.wavesplatform.lang.v1.ContractLimits
+import com.wavesplatform.network
+import com.wavesplatform.state.{BinaryDataEntry, BooleanDataEntry, IntegerDataEntry, StringDataEntry}
+import com.wavesplatform.test.FlatSpec
+import com.wavesplatform.transaction.{DataTransaction, TxHelpers}
+
+class DataTransactionNewLimitsSpec extends FlatSpec with WithDomain {
+  def generateMaxAllowed(limit: Int): (BooleanDataEntry, IntegerDataEntry, StringDataEntry, Seq[BinaryDataEntry]) = {
+    val KeySize = 1
+    val ArbitraryEntriesCount = 10
+    val bool    = BooleanDataEntry("1", value = true)
+    val int     = IntegerDataEntry("2", Long.MaxValue)
+
+    val bytesOrStrEntryLimit = (limit - KeySize * 2 - 8 - 1 - (KeySize * ArbitraryEntriesCount)) / ArbitraryEntriesCount
+    val str                  = StringDataEntry("3", String.valueOf(Array.fill(bytesOrStrEntryLimit)('A')))
+    val bytes                = (1 until ArbitraryEntriesCount).map(i => BinaryDataEntry((3 + i).toString, ByteStr(Array.fill(bytesOrStrEntryLimit)('A'.toByte))))
+    (bool, int, str, bytes)
+  }
+
+  "Data transaction" should "handle new limits" in withDomain(DomainPresets.RideV6) { d =>
+    d.helpers.creditWavesToDefaultSigner()
+
+    val (bool, int, str, bytes) = generateMaxAllowed(DataTransaction.MaxBytes)
+    val dataEntries             = Seq(bool, int, str) ++ bytes
+    val dataTransaction         = TxHelpers.dataWithMultipleEntries(TxHelpers.defaultSigner, dataEntries)
+    d.appendAndAssertSucceed(dataTransaction)
+    
+    val networkMessage = network.TransactionSpec.serializeData(dataTransaction)
+    networkMessage.length shouldBe <= (network.TransactionSpec.maxLength)
+
+    val protoNetworkMessage = network.PBTransactionSpec.serializeData(dataTransaction)
+    protoNetworkMessage.length shouldBe <= (network.PBTransactionSpec.maxLength)
+  }
+
+  it should "handle 400 bytes key" in withDomain(DomainPresets.RideV6) { d =>
+    val dataEntry = StringDataEntry("a" * 400, "test")
+    val dataTransaction = TxHelpers.dataWithMultipleEntries(TxHelpers.defaultSigner, Seq(dataEntry))
+
+    d.helpers.creditWavesToDefaultSigner()
+    d.appendAndAssertSucceed(dataTransaction)
+    d.blockchain.accountData(TxHelpers.defaultAddress, "a" * 400) shouldBe Some(dataEntry)
+  }
+
```

Comment:
Add negative tests (just in case): 
- key > 400 bytes
- value > 32767 bytes
- overall data > 165835 bytes (with keys <= 400 and data <= 32767)